sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -359,16 +425,23 @@ AccessToken TokenCredentialImpl::ParseToken(
         for (auto const& parse : {
                  std::function<DateTime(std::string const&)>([&](auto const& s) {
                    // 'expires_on' as RFC3339 date string (absolute timestamp)
-                   return DateTime::Parse(s + tzOffsetStr, DateTime::DateFormat::Rfc3339);
+                   auto dateTime = DateTime::Parse(s + tzOffsetStr, DateTime::DateFormat::Rfc3339);
+                   return proactiveRenewal ? GetProactiveRenewalDateTime(
+                              PosixTimeConverter::DateTimeToPosixTime(dateTime))
+                                           : dateTime;
                  }),
-                 std::function<DateTime(std::string const&)>([](auto const& s) {
+                 std::function<DateTime(std::string const&)>([&](auto const& s) {
                    // 'expires_on' as numeric string (posix time representing an absolute timestamp)
-                   return PosixTimeConverter::PosixTimeToDateTime(
-                       ParseNumericExpiration(s, MaxPosixTimestamp));
+                   auto value = ParseNumericExpiration(s, MaxPosixTimestamp);
+                   return proactiveRenewal ? GetProactiveRenewalDateTime(value)
+                                           : PosixTimeConverter::PosixTimeToDateTime(value);
                  }),
-                 std::function<DateTime(std::string const&)>([](auto const& s) {
+                 std::function<DateTime(std::string const&)>([&](auto const& s) {
                    // 'expires_on' as RFC1123 date string (absolute timestamp)
-                   return DateTime::Parse(s, DateTime::DateFormat::Rfc1123);
+                   auto dateTime = DateTime::Parse(s, DateTime::DateFormat::Rfc1123);
+                   return proactiveRenewal ? GetProactiveRenewalDateTime(
+                              PosixTimeConverter::DateTimeToPosixTime(dateTime))
+                                           : dateTime;
```

Comment:
FWIW, line 455 is insanely suspect to me - we catch the exception and discard it with a comment that we might throw later - but the exception is never captured for rethrowing.

Reply:
I agree. I followed the existing coding pattern in the file, but it can be written more cleanly.

The exception is being swallowed and later a "cleaner" exception is thrown with more user-relevant diagnostics.